### PR TITLE
Preferential Alpha Compositing in Plots

### PR DESCRIPTION
**Pull request by wbinventor**
This PR adds a new `Plot.highlight_domains(...)` routine to the Python API which builds upon the automated color schemes introduced in PR #503. The routine can be used to apply [alpha compositing](https://en.wikipedia.org/wiki/Alpha_compositing) to partially shade all domains (cells or materials) other than those specified by the user. The resulting effect is to "highlight" those domains which are left out of the shading process.

An example of how this routine may be used is shown below for a fuel assembly in the BEAVRS benchmark with OpenCG local neighbor symmetry (LNS) identification algorithm applied. In this case, I used the built-in "grey" background with an alpha of 0.1

*BEAVRS 1.6% enr. fuel assembly w/o BAs*

*With OpenCG LNS Identifers Applied*

*Highlight LNS ID=119*

*Highlight LNS ID=185*

*Highlight LNS ID=251*

*Highlight LNS ID=317*

*Highlight LNS ID=440*

### CODE REVIEW

File: openmc/plots.py [hunk 1]
```diff
@@ -261,6 +261,97 @@ def __repr__(self):
...
+        cv.check_greater_than('alpha', alpha, 0., equality=True)
+        cv.check_less_than('alpha', alpha, 1., equality=True)
+
+        # Get a background (R,G,B) tuple to apply in alpha compositing
+        if isinstance(background, basestring):
+            if background == 'white':
+                background = (255, 255, 255)
+            elif background == 'black':
+                background = (0, 0, 0)
+            elif background == 'grey':
```
Comment (paulromano) [inline openmc/plots.py]: If I'm not mistaken, "gray" is more accepted in American English.

File: openmc/plots.py [hunk 2]
```diff
@@ -261,6 +261,97 @@ def __repr__(self):
...
+
+
+    def highlight_domains(self, geometry, domains, seed=1,
+                          alpha=0.5, background='grey'):
+        """Use alpha compositing to highlight one or more domains in the plot.
+
+        This routine generates a color scheme and applies alpha compositing
+        to make all domains except the highlighted ones partially transparent.
+
+        Params
```
Comment (paulromano) [inline openmc/plots.py]: Parameters

File: openmc/plots.py [hunk 3]
```diff
@@ -349,6 +440,51 @@ def remove_plot(self, plot):
...
```
Comment (paulromano) [inline openmc/plots.py]: Parameters

File: openmc/plots.py [hunk 4]
```diff
@@ -261,6 +261,97 @@ def __repr__(self):
...
+            b = np.random.randint(0, 255)
+            self.col_spec[domain] = (r, g, b)
+
+
+    def highlight_domains(self, geometry, domains, seed=1,
+                          alpha=0.5, background='grey'):
+        """Use alpha compositing to highlight one or more domains in the plot.
+
+        This routine generates a color scheme and applies alpha compositing
+        to make all domains except the highlighted ones partially transparent.
```
Comment (paulromano) [inline openmc/plots.py]: To be precise I would say "appear partially transparent". Sorry to nitpick.

Comment (paulromano) [inline openmc/plots.py]: You know there's not much wrong with your code once I start picking at your English!

**Comment by paulromano** [comment]
@wbinventor Cool eye candy! This and #503 should be ready to go once the minor comments are resolved.

File: openmc/plots.py (hunk 1 above)
Comment (wbinventor) [inline openmc/plots.py]: Haha, it was the irish in me coming out in my code, that's all.

**Comment by wbinventor** [comment]
All comments have been addressed

**Comment by paulromano** [comment]
Thanks @wbinventor!

**Pull request merged**